packages/smooth_app/lib/pages/product/common/product_list_popup_items.dart

```diff
@@ -216,6 +222,112 @@ class ProductListPopupOpenInWeb extends ProductListPopupItem {
+      appLocalizations.product_list_export;
+
+  @override
+  IconData getIconData() => Icons.download;
+
+  @override
+  ProductListPopupMenuEntry getEntry() => ProductListPopupMenuEntry.export;
+
+  @override
+  bool isDestructive() => false;
+
+  @override
+  Future<ProductList?> doSomething({
+    required final ProductList productList,
+    required final LocalDatabase localDatabase,
+    required final BuildContext context,
+  }) async {
+    final String listName = ProductQueryPageHelper.getProductListLabel(
+      productList,
+      AppLocalizations.of(context),
+    );
+    final String fileName =
+        "${listName.replaceAll(' ', '-').toLowerCase()}_${DateTime.now().millisecondsSinceEpoch}.csv";
+    final StringBuffer csv = StringBuffer('Barcode,Name,Brand\n');
+
+    for (final String barcode in productList.barcodes) {
+      final Product? product = await DaoProduct(localDatabase).get(barcode);
+      if (product != null) {
+        final String name = (product.productName ?? '').replaceAll(';', '');
+        final String brand = (product.brands ?? '').replaceAll(';', '');
+        csv.write('"$barcode";"$name";"$brand"\n');
+      }
+    }
+
+    unawaited(
+      Share.shareXFiles(
+        <XFile>[
+          XFile.fromData(
+            utf8.encode(csv.toString()),
+            name: '$fileName.csv',
+            mimeType: 'text/csv',
+          ),
+        ],
+        fileNameOverrides: <String>['$fileName.csv'],
+      ),
+    );
+
+    return null;
+  }
+}
+
+/// Popup menu item for the product list page: import list.
+class ProductListPopupImport extends ProductListPopupItem {
+  @override
+  String getTitle(final AppLocalizations appLocalizations) =>
+      appLocalizations.product_list_import;
+
+  @override
+  IconData getIconData() => Icons.upload;
+
+  @override
+  ProductListPopupMenuEntry getEntry() => ProductListPopupMenuEntry.import;
+
+  @override
+  bool isDestructive() => false;
+
+  @override
+  Future<ProductList?> doSomething({
+    required final ProductList productList,
+    required final LocalDatabase localDatabase,
+    required final BuildContext context,
+  }) async {
+    FilePicker.platform.pickFiles(
+      type: FileType.custom,
+      allowedExtensions: <String>['csv'],
+    ).then((final FilePickerResult? result) {
+      if (result == null) {
+        return;
+      }
+      final File file = File(result.files.single.path!);
+      final String content = file.readAsStringSync();
+      final List<String> lines = content.split('\n');
+      final List<String> barcodes = <String>[];
+      for (final String line in lines) {
+        final List<String> parts = line.split(';');
+        if (parts.length < 3) {
+          continue;
+        }
+        final String barcode = parts[0].replaceAll('"', '');
+        barcodes.add(barcode);
+      }
+      DaoProductList(localDatabase).bulkSet(
+        productList,
+        barcodes,
+        include: true,
+      );
+    });
+
+    return null;
```

Comment:
You've just imported data, so I guess the list has changed.
In that case you don't return `null`.